### PR TITLE
added .check()

### PR DESCRIPTION
**Pull request by timbowhite**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -167,7 +171,8 @@ function lock(file, options, compromised, callback) {
         update: null,
         realpath: true,
         retries: 0,
-        fs: fs
+        fs: fs,
+        check: false
```
Comment (satazor) [inline index.js]: I think this is a typo, can this be removed?

File: index.js [hunk 2]
```diff
@@ -303,6 +308,42 @@ function unlockSync(file, options) {
     }
 }
 
+// checks if file is currently locked. callback with boolean, true if locked, false if unlocked.
```
Comment (satazor) [inline index.js]: Remove this command.

**Comment by satazor** [comment]
The PR looks good, please review my comments.

Also:
- Please add `.checkSync` method and tests (see how I've implemented `lockSync` etc)
- There are some linting errors; I changed from jshint to eslint and updated some code (you may need to resolve some minor conflicts too)

**Comment by timbowhite** [comment]
OK should be updated.  Let me know if you'd prefer I rebase/squash my 2 commits to a single commit.

**Comment by satazor** [comment]
@timbowhite I will rebase it, don't worry. I still see some linting errors, did you run `npm i && npm run lint`?

**Comment by timbowhite** [comment]
> I still see some linting errors, did you run npm i && npm run lint?

I'm on node v0.10.42, and eslint doesn't support node < 4 after version 2.13.1.

Trying the ```eslint '{*.js,lib/**/*.js,test/**/*.js}' --ignore-pattern=test/coverage``` with eslint 2.13.1, I get this error:
```
Error: Cannot find module '@satazor/eslint-config/addons/node'
```

**Comment by satazor** [comment]
Oh, why are you using such old version?

**Comment by timbowhite** [comment]
Node v0.10.x is the current version released in the default repos for Centos 6 (and Ubuntu 14 too I believe).

I'll use nvm to install node v4 and see if i can get eslint working.

**Comment by timbowhite** [comment]
OK lint issues should be fixed, used node v6.5.0 and eslint v3.5.0.

**Pull request merged**